extensions/typescript-language-features/src/languageFeatures/completions.ts

```diff
@@ -536,15 +535,47 @@ class TypeScriptCompletionItemProvider implements vscode.CompletionItemProvider<
 			useFuzzyWordRangeLogic: this.client.apiVersion.lt(API.v390),
 		};
 
+		let includesPackageJsonImport = false;
 		const items: MyCompletionItem[] = [];
 		for (let entry of entries) {
 			if (!shouldExcludeCompletionEntry(entry, completionConfiguration)) {
 				items.push(new MyCompletionItem(position, document, entry, completionContext, metadata));
+				// @ts-expect-error - remove after TS 4.0 protocol update
+				includesPackageJsonImport = !!entry.isPackageJsonImport;
 			}
 		}
+		if (duration !== undefined) {
+			this.logCompletionsTelemetry(duration, response, includesPackageJsonImport);
+		}
 		return new vscode.CompletionList(items, isIncomplete);
 	}
 
+	private logCompletionsTelemetry(
+		duration: number,
+		response: ServerResponse.Response<Proto.CompletionInfoResponse> | undefined,
+		includesPackageJsonImport?: boolean
+	) {
+		/* __GDPR__
+			"completions.execute" : {
+				"duration" : { "classification": "SystemMetadata", "purpose": "FeatureInsight" },
+				"type" : { "classification": "SystemMetadata", "purpose": "FeatureInsight" },
+				"count" : { "classification": "SystemMetadata", "purpose": "FeatureInsight" },
+				"updateGraphDurationMs" : { "classification": "SystemMetadata", "purpose": "FeatureInsight" },
+				"includesPackageJsonImport" : { "classification": "SystemMetadata", "purpose": "FeatureInsight" },
+				"${include}": [
+					"${TypeScriptCommonProperties}"
+				]
+			}
+		*/
+		this.telemetryReporter.logTelemetry('completions.execute', {
+			duration: duration,
+			type: response?.type ?? 'unknown',
+			count: response?.type === 'response' && response.body ? response.body.entries.length : 0,
+			updateGraphDurationMs: response?.type === 'response' ? response.performanceData?.updateGraphDurationMs : undefined,
+			includesPackageJsonImport: includesPackageJsonImport ? 'true' : undefined,
```

Comment:
Is there a convention for modeling boolean properties? Boolean values aren’t allowed in the `TelemetryProperties` type.

Reply:
There's this: https://github.com/microsoft/vscode/blob/afdb5ec6e06bea182d7fdcde2318f391c8d55f6c/extensions/typescript-language-features/src/typescriptServiceClient.ts#L864

Reply:
Which produces `'true'` or `'false'` 👍. I suspect it’s just as easy to distinguish between `'true'` and `undefined` as between `'true'` and `'false'` so I’m happy with this.